src/bindings/python/flux/future.py

```diff
@@ -48,12 +48,22 @@ def continuation_callback(c_future, opaque_handle):
         flux_handle.reactor_stop_error()
 
     finally:
-        _THEN_HANDLES[py_future] -= 1
-        if _THEN_HANDLES[py_future] <= 0:
-            # allow future object to be garbage collected now that all
-            # registered callbacks have completed
-            py_future.cb_handle = None
-            del _THEN_HANDLES[py_future]
+        if py_future in _THEN_HANDLES:
+            _THEN_HANDLES[py_future] -= 1
+            if _THEN_HANDLES[py_future] <= 0:
+                #
+                #  Note, a multiply fulfilled future which is not reset
+                #  before leaving the then_cb() will end up here, since a
+                #  call to reset() is the only thing that incrementsthe
```

Comment:
need a space between increments and the

Reply:
```suggestion
                #  call to reset() is the only thing that increments the
```

Reply:
Ah, good catch, thanks!

Reply:
Ok fixed, thanks!

Reply:
I wonder why the typos module didn't catch this one?